Add cgroup bind mount by default

hqhq opened the pull request:
Libcontainer already supported mount container's own cgroup into
container, with this patch, we can see container's own cgroup info
in container.

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

calavera commented:
ping @crosbymichael, @LK4D4

hqhq commented:
ping

LK4D4 commented:
ping @crosbymichael @ewindisch 
Do we want to expose this info to containers?

ewindisch commented:
I don't know of a direct security impact here, although it does open the surface area in that vulnerabilities in the kernel could potentially allow remounting this directory.

More generally, however, I question if we should break the fourth-wall here. There is a line of thought that containers need not know that they're containers, nor know what their cgroup configuration is. What is the value of this to users / developers?

hqhq commented:
@ewindisch One thing for clear is that this PR doesn't make containers know that they're containers, but make them more like they are hosts which have no sub cgroups. Maybe the RO makes it a bit weird.

icecrime commented:
*Collective review with @jfrazelle @crosbymichael @calavera*

We're fine making this a default as long as the security team gives a green flag on this (which apparently @ewindisch already does). Ping @diogomonica @NathanMcCauley for other opinions.

NathanMcCauley commented:
LGTM

LK4D4 commented:
LGTM

icecrime commented:
LGTM

icecrime commented:
Do we need docs anywhere for this?

hqhq commented:
@icecrime It only affect applications in the container, and there are no UI yet, I don't think we have docs for this, like fs specs in the container, do we?

crosbymichael commented:
@hqhq this looks good.  Do you think you could add a test to ensure that it is RO and we don't regress on that point?

ewindisch commented:
LGTM

hqhq committed:
Add cgroup bind mount by default

Libcontainer already supported mount container's own cgroup into
container, with this patch, we can see container's own cgroup info
in container.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq commented:
@crosbymichael Test case added, PTAL.

crosbymichael commented:
LGTM

dmcgowan commented:
Having an issue when running Docker in Docker and bisected the problem it to this commit.

```
Cannot start container <omitted>: [8] System error: Error found less than 3 fields post '-' in "262 261 0:85 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime - tmpfs  rw,seclabel"
```

jfrazelle commented:
This is a known bug on master it's fixed in runc 

> On Jul 15, 2015, at 17:37, Derek McGowan <notifications@github.com> wrote:
> 
> Having an issue when running Docker in Docker and bisected the problem it to this commit.
> 
> Cannot start container <omitted>: [8] System error: Error found less than 3 fields post '-' in "262 261 0:85 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime - tmpfs  rw,seclabel"
> —
> Reply to this email directly or view it on GitHub.
>
